test/one/profiler/test/TestDeclaration.java

```diff
@@ -0,0 +1,137 @@
+            for (File file : files) {
+                if (file.isDirectory()) {
+                    allTestDirs.add(file.getName());
+                }
+            }
+        }
+
+        return new TestDeclaration(allTestDirs, filters, skipFilters);
+    }
+
+    private static List<RunnableTest> getRunnableTests(String dir) {
+        String className = "test." + dir + "." + Character.toUpperCase(dir.charAt(0)) + dir.substring(1) + "Tests";
+        try {
+            List<RunnableTest> rts = new ArrayList<>();
+            for (Method m : Class.forName(className).getMethods()) {
+                for (Test t : m.getAnnotationsByType(Test.class)) {
+                    rts.add(new RunnableTest(m, t));
+                }
+            }
+            return rts;
+        } catch (ClassNotFoundException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public List<RunnableTest> getRunnableTests() {
+        return allDirs.stream()
+            .flatMap(g -> getRunnableTests(g).stream())
+            .sorted(Comparator.comparing(RunnableTest::testName))
+            .collect(Collectors.toList());
+    }
+
+    public boolean matches(Method m) {
+        String name = m.getDeclaringClass().getSimpleName() + '.' + m.getName();
+
+        if (includeGlobs.isEmpty() || includeGlobs.stream().anyMatch(f -> f.matcher(name).matches())) {
+            return skipGlobs.isEmpty() || !skipGlobs.stream().anyMatch(f -> f.matcher(name).matches());
+        }
+
+        return false;
+    }
+
+    private static Pattern filterFrom(String s) {
+        if (s.startsWith("*") && s.endsWith("*")) {
+            // contains.
+            return Pattern.compile(".*" + Pattern.quote(s.substring(1, s.length() - 1)) + ".*");
+        }
+        if (s.startsWith("*")) {
+            // ends with.
+            return Pattern.compile(".*" + Pattern.quote(s.substring(1)) + "$");
+        }
+        if (s.endsWith("*")) {
+            // starts with.
+            return Pattern.compile("^" + Pattern.quote(s.substring(0, s.length() - 1)) + ".*");
+        }
+
+        // equals
+        return Pattern.compile("^" + Pattern.quote(s) + "$");
+    }
+
+    private List<Pattern> filterFromGlobs(List<String> globs) {
+        Set<String> result = new HashSet<>();
+        for (String g : globs) {
+            if (allDirs.contains(g)) {
+                // all lowercase, folder name.
+                result.add(g.substring(0, 1).toUpperCase() + g.substring(1).toLowerCase() + "Tests.*");
+            } else if (g.contains(".") || g.contains("*")) {
+                result.add(g);
+            } else if (Character.isUpperCase(g.charAt(0))) {
+                // Looks like class name.
+                result.add(g.substring(0, 1).toUpperCase() + g.substring(1).toLowerCase() + ".*");
+            } else if (Character.isLowerCase(g.charAt(0))) {
+                // Looks like method name.
+                result.add("*." + g.toLowerCase());
```

Comment:
Why do we convert it to lowercase? AFAICS, we use case sensitive patterns.

Reply:
It was a misunderstanding, correcting to `result.add("*." + g);`.